src/app/file-manager/models/export-to-gcs-item.ts

```diff
@@ -0,0 +1,59 @@
+import { Type } from './type';
+import { EntityStatus } from './entity-status';
+
+export class ExportToGCSItem {
+  id: string;
+  name: string;
+  displayName: string;
+  created: Date;
+  updated: Date;
+  size: number;
+  mediaLink: string;
+  path: string;
+  preserveStructure: boolean;
+  destination: string;
+  progress: number;
+  status: EntityStatus;
+  transferred: number;
+  remaining: number;
+  bucketName: string;
+  workspaceName: string;
+  prefix: string;
+  delimiter: string;
+  selected: boolean;
+  indeterminate: boolean;
+  namespace: string;
+
+  constructor(id, name, created, updated, size, mediaLink, path, destination,
+    status, bucketName, prefix, delimiter, preserveStructure,
+    workspaceName, displayName, namespace) {
+
+    this.id = id;
+    this.name = name;
+    this.created = created;
+    this.updated = updated;
+    this.size = size;
+    this.mediaLink = mediaLink;
+    this.path = path;
+    this.preserveStructure = preserveStructure;
+    this.destination = destination;
+    this.status = status;
+    this.bucketName = bucketName;
+    this.prefix = prefix;
+    this.delimiter = delimiter;
+    this.selected = false;
+    this.indeterminate = false;
+    this.workspaceName = workspaceName;
+    this.displayName = displayName;
+    this.setProgressData();
+    this.namespace = namespace;
+  }
+
+  setProgressData() {
```

Comment:
A more OO solution would put common items into a shared class, or a common base class, to avoid duplication.

Reply:
Do you mean  in xxxx-item.ts classes ?- We had a single item class for all, and later I created a class for each item: download, upload, export to gcs and export to s3, because data needs were different and we saw the need to split reducers to improve performance. We have not finished yet the cleaning of items properties after the refactoring, we expected to continue it in some moment after release. You will find similar things in other classes as well. We tried to keep features as decoupled as possible until we understand well use cases and data needs for each case. We migth create a story to refactor code as you suggest.

Reply:
OK